src/sail_c_backend/c_backend.mli

```diff
@@ -51,6 +51,18 @@ open Type_check
 
 (** Global compilation options *)
 
+(** If set generate a C++ struct for the model instead of global C functions/variables. *)
+val opt_cpp : bool ref
```

Comment:
I wonder if these could be an option in the `CODEGEN_CONFIG` module type? Would avoid adding any mutable variables, as I was trying to move away from these.